### PR TITLE
Rest api - refactor item view model to root and child, and overwrite logic

### PR DESCRIPTION
**Pull request by hellosnow**
1. For rest api, current view model is not fit for tag. For example, root's tag is list of RestApiTagViewModel, while each api is list of string. Split view model to root and child item view model.
2. Rewrite overwrite logic both for root and child.

@ansyral @chenkennt @superyyrrzz @vwxyzh @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    using System.Composition;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Build.Common;
+    using Microsoft.DocAsCode.Build.RestApi.ViewModels;
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    [Export(nameof(RestApiDocumentProcessor), typeof(IDocumentBuildStep))]
+    public class ApplyOverwriteDocumentForRestApiRootChild : ApplyOverwriteDocument
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs]: why not merge RestApiRootChild and RestApiRoot into one?

File: src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                var rootItem = (RestApiRootItemViewModel)articleModel.Content;
+                if (rootItem.Children.Any(c => c.Uid == uid))
+                {
+                    return rootItem.Children.Where(c => c.Uid == uid);
+                }
+
+                return EmptyChildItemList;
+            }));
+
+        protected override void ApplyOvewriteDocument(IHostService host, List<FileModel> od, string uid, List<FileModel> articles)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs]: typo, should be ApplyOverwriteDocument

File: src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs (hunk 1 above)
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApiChild.cs]: I got your point, should merge into one class and judge the logic inside ApplyOvewriteDocument

---
In reply to: [64860219](https://github.com/dotnet/docfx/pull/336#discussion_r64860219) [](ancestors = 64860219)

**Comment by vwxyzh** [comment]
👍

**Comment by superyyrrzz** [comment]
:+1:

**Comment by hellosnow** [comment]
Merge #336

**Pull request merged**